cds_migrator_kit/videos/weblecture_migration/transform/transform.py

```diff
@@ -0,0 +1,236 @@
+        return
+
+    def _version_id(self, entry):
+        return
+
+    def _index(self, entry):
+        return
+
+    def _pids(self, entry):
+        return
+
+    def _bucket_id(self, json_entry):
+        return
+
+    def _media_bucket_id(self, entry):
+        return
+
+    def _files(self, record_dump):
+        """Transform the files of a record."""
+        raise NotImplementedError("_files is not implemented for this class.")
+
+    def _media_files(self, entry):
+        return
+
+    def _metadata(self, entry):
+        """Transform the metadata of a record."""
+
+        def reformat_date(json_data):
+            """Reformat the date for the cds-videos data model."""
+            dates = json_data.get("date", [])
+            dates_set = {date for date in dates if date is not None}
+
+            if len(dates_set) == 1:  # Should be only one value
+                return next(iter(dates_set))  # Get the single date from the set
+            if len(dates_set) > 1:
+                return next(iter(dates_set))  # return the first
+
+            # print("No valid date found in record: {}.".format(json_data.get('recid')))
+            raise UnexpectedValue(
+                "No valid date found in record: {}.".format(json_data.get("recid"))
+            )
+
+        def description(json_data):
+            """Reformat the description for the cds-videos data model."""
+            if not json_data.get("description"):
+                # print("\tNo description found, empty string added")
+                return json_data.get("title").get("title")
+            return json_data.get("description")
+
+        metadata = {
+            "title": entry["title"],
+            "description": description(entry),
+            "contributors": entry.get("contributors"),
+            "languages": entry.get("language"),
+            "date": reformat_date(entry),
+        }
+        # filter empty keys
+        return {k: v for k, v in metadata.items() if v}
+
+    def _custom_fields(self, entry):
+        """Transform the custom fields of a record."""
+        return
+
+    def transform(self, entry):
+        """Transform a record single entry."""
+        record_dump = CDSRecordDump(data=entry, dojson_model=videos_migrator_marc21)
+        migration_logger = RDMJsonLogger()
+
+        record_dump.prepare_revisions()
+        timestamp, json_data = record_dump.latest_revision
+
+        migration_logger.add_record(json_data)
+        record_json_output = {
+            "created": self._created(json_data),
+            "updated": self._updated(record_dump),
+            "metadata": self._metadata(json_data),
+        }
+
+        return {
+            "created": self._created(json_data),
+            "updated": self._updated(record_dump),
+            "recid": self._recid(record_dump),
+            "json": record_json_output,
+            # keep the original extracted entry for storing it
+            "_original_dump": entry,
```

Comment:
note: this would need to implement a dedicated table in cds-videos to save all the original extracted migrated records. See https://github.com/CERNDocumentServer/cds-rdm/blob/master/site/cds_rdm/legacy/models.py#L21

Can you create an issue for that so we don't forget?